### ISSUE TITLE
`nimble dump`: Different output between `stable` and `devel` versions

### ISSUE THREAD

**Issue by CrazyManLabs** [nim-lang/nimble#941]
I'm trying out latest versions of `nim` and `nimble` after reading [this forum post](https://forum.nim-lang.org/t/8404).
As I'm using `choosenim` - I'm switching between versions using `choosenim stable` and `choosenim devel` commands.

One of my build scripts uses `nimble dump` command to get information about package without the need to parse/evaluate `*.nimble` file.
To be exact - I'm trying to detect, if current package will generate dynamic library or executable by checking extension of each entry in `bin` property, returned from `nimble dump` (some customizations of `*.nimble` file are required to make it work; this is not the whole task, which I'm trying to solve, but the basic idea is as described).

I have these customizations in `*.nimble` file:
```nim
# ...

bin = @[projectName()]

import std/os
for name in bin:
  namedBin[name] = name.toDll()

# ...
```

Also here is my `config.nims` (to make it clear what I'm trying to build):
```nim
switch("app", "lib")

# ...
```

I'm using these constructs, because:
- I do not want to change contents of `*.nimble` file (line with `bin = @[...]`) itself whenever I rename package (as during development it is hard to name things right from the first attempt :) );
- I want to generate correctly-named binary (as I'm building dynamic library and not an executable, but `nimble build` generates binary with same filename for both cases without this modification).

I see one key difference in generated output between `stable` and `devel` versions, which prevents my script from doing its thing:
- when using `stable` version: `nimble dump` will give information about `bin` list respecting `namedBin` overrides:
  ```console
  $ choosenim stable
     Switched to Nim 1.4.8

  $ nimble dump
  name: "<some_package_name>"
  # ...
  bin: "lib<some_package_name>.so"
  # ...
  ```
- when using `devel` version: `nimble dump` will give information about `bin` list without any adaptations, made by `namedBin` overrides:
  ```console
  $ choosenim devel
     Switched to Nim #devel
    
  $ nimble dump
  name: "<some_package_name>"
  # ...
  bin: "<some_package_name>"
  # ...
  ```

Is this a regression/bug or is this expected behavior?
If this is expected behavior - is there other way of getting information from `namedBin` property without parsing/evaluating `*.nimble` file "by hand"?
_Also one more question (not related to the issue, but still ...): is there a way to tell `nimble` to generate correctly-named binary depending on its type (`exe` or `dll`) without such customizations?_

Here are some details about used tools:
```console
$ choosenim --version
choosenim v0.7.4 (2020-10-20 17:30:52) [linux/amd64]

$ choosenim stable
   Switched to Nim 1.4.8

$ nimble --version
nimble v0.13.1 compiled at 2021-05-25 00:49:34
git hash: couldn't determine git hash

$ choosenim devel
   Switched to Nim #devel

$ nimble --version
nimble v0.13.1 compiled at 2021-09-15 00:51:39
git hash: couldn't determine git hash
```

And here are some details about environment (if this is needed):
- KDE Neon 5.22.5 (based on Ubuntu 20.04 (focal))
- Linux 5.11.0-34-generic #&#8203;36~20.04.1-Ubuntu SMP Fri Aug 27 08:06:32 UTC 2021 x86_64 GNU/Linux

**Comment by dom96** [comment]
@bobeff this looks like a regression

**Comment by bobeff** [comment]
@CrazyManLabs @dom96 Yes, it was a regression. I fixed it in 00d088e.
